docs/developer-docs/developer-tools/cli-tools/dfx-json.mdx

```diff
@@ -0,0 +1,193 @@
+
+## Default `dfx.json` file
+
+By default, when a project is created with `dfx new <project-name>`, and the options `Motoko` and `No frontend canister` are selected, the default `dfx.json` generated for the project will contain the following:
+
+```json
+{
+  "canisters": {
+    "example_backend": {
+      "main": "src/example_backend/main.mo",
+      "type": "motoko"
+    }
+  },
+  "defaults": {
+    "build": {
+      "args": "",
+      "packtool": ""
+    }
+  },
+  "output_env_file": ".env",
+  "version": 1
+}%
+```
+
+This configuration contains the following components:
+
+- `canisters`: Defines the canister names and settings for the project. In this default example, a single canister called `example_backend` is defined.
+
+- `example_backend`: Defines the name of the canister in this project.
+
+- `main`: The primary source code file for the canister that serves as the canister's entrypoint. By default, for Motoko canisters this file is `main.mo`. For Rust canisters,
+
+- `type`: Defines the type of the canister. Supported types are `motoko`, `rust`, `assets`, `pull` and `custom`. The Motoko and Rust languages are supported through the IC SDK by default. `custom` type refers to canisters written in languages other than Motoko or Rust.
+
+- `defaults`: Default settings for the project's build.
+
+- `build`: Build process configuration.
+
+- `args`: Arguments for the `packtool` setting.
+
+- `packtool`: Configures a package management tool for your canister's dependencies. Examples include `mops` and `vessel` for Motoko.
+
+- `output_env_file`: Environment variables will be written to this file without overwriting any user-defined variables if the file already exists.
+
+- `version`: Refers to the iteration version for this `dfx.json` file. By default, versions of `dfx.json` start at `version: 1`.
+
+## Custom canister definitions
+
+To define a canister to use a language other than `Motoko` or `Rust`, or to use a custom Wasm file, you can set the canister as `"type": "custom"` with the following additional arguments:
+
+```json
+"canisters": {
+"example_backend": {
+    "build": "npx azle example_backend",
+    "candid": "src/example_backend/example_backend.did",
+    "gzip": true,
+    "main": "src/example_backend/src/index.ts",
+    "type": "custom",
+    "wasm": ".azle/example_backend/example_backend.wasm"
+}
+}
+```
+
+- `build`: The command that will build the canister.
+
+- `candid`: The file path to the canister's Candid interface definition file.
+
+- `gzip`: `True` or `False` if Wasm gzip should be used.
```

Comment:
nit: true and false should be lowercase as JSON doesn't recognize capitalized bools